src/components/collectible-icon/collectible-icon.tsx

```diff
@@ -11,92 +17,105 @@ import {
-  { type: 'displayUri', uri: formatImgUri, field: 'displayUri' },
-  { type: 'artifactUri', uri: formatImgUri, field: 'artifactUri' },
-  { type: 'thumbnailUri', uri: formatImgUri, field: 'thumbnailUri' }
-];
-
-const collectibleThumbnailLoadStrategy: Array<LoadStrategy> = [
-  { type: 'objktMed', uri: formatCollectibleObjktMediumUri, field: 'assetSlug' }, // gif
-  { type: 'objktArtifact', uri: formatCollectibleObjktArtifactUri, field: 'artifactUri' },
-  { type: 'displayUri', uri: formatImgUri, field: 'displayUri' },
-  { type: 'artifactUri', uri: formatImgUri, field: 'artifactUri' },
-  { type: 'thumbnailUri', uri: formatImgUri, field: 'thumbnailUri' }
-];
-
-interface MetadataFormats {
-  dimensions: {
-    unit: string;
-    value: string;
-  };
-  fileName: string;
-  fileSize: number;
-  mimeType: string;
-  uri: string;
-}
-
-type ImageRequestObject = {
-  assetSlug: string;
-  displayUri?: string;
-  artifactUri?: string;
-  thumbnailUri?: string;
-  formats?: Array<MetadataFormats>;
-};
-
-const getFirstFallback = (
-  strategy: Array<LoadStrategy>,
-  currentState: Record<string, boolean>,
-  metadata: ImageRequestObject
-): LoadStrategy => {
-  for (const strategyItem of strategy) {
-    const isArtifactUri = isDefined(metadata.artifactUri) && strategyItem.field === 'artifactUri';
-    const isDisplayUri = isDefined(metadata.displayUri) && strategyItem.field === 'displayUri';
-    const isThumbnailUri = isDefined(metadata.thumbnailUri) && strategyItem.field === 'thumbnailUri';
-    const isObjktMed =
-      ((isDefined(metadata.formats) && metadata.formats.some(x => x.mimeType === 'image/gif')) ||
-        !isDefined(metadata.formats)) &&
-      strategyItem.type === 'objktMed';
-    if ((isArtifactUri || isDisplayUri || isThumbnailUri || isObjktMed) && !currentState[strategyItem.type]) {
-      return strategyItem;
-    }
-  }
-
-  return strategy[0];
-};
-
 export const CollectibleIcon: FC<CollectibleIconProps> = ({
   collectible,
   size,
-  iconSize = CollectibleIconSize.SMALL
+  iconSize = CollectibleIconSize.SMALL,
+  mime,
+  objktArtifact,
+  setScrollEnabled
 }) => {
+  const [isLoading, setIsLoading] = useState(true);
+
+  const isBigIcon = iconSize === CollectibleIconSize.BIG;
   const styles = useCollectibleIconStyles();
-  const actualLoadingStrategy =
-    iconSize === CollectibleIconSize.SMALL ? collectibleThumbnailLoadStrategy : collectibleBigLoadStrategy;
-  const [isLoadingFailed, setIsLoadingFailed] = useState(
-    actualLoadingStrategy.reduce<Record<string, boolean>>((acc, cur) => ({ ...acc, [cur.type]: false }), {})
-  );
-  const assetSlug = `${collectible?.address}_${collectible?.id}`;
+  const assetSlug = `${collectible.address}_${collectible.id}`;
+
+  const initialFallback = useMemo(() => {
+    if (isDefined(collectible.artifactUri) && isBigIcon) {
+      return formatCollectibleObjktArtifactUri(collectible.artifactUri);
+    }
+
+    return formatCollectibleObjktMediumUri(assetSlug);
+  }, []);
```

Comment:
don't you need deps?

Reply:
initialFallback used only once for initializing state, so probably not